src/msw/textctrl.cpp

```diff
@@ -517,6 +519,10 @@ bool wxTextCtrl::MSWCreateText(const wxString& value,
                 }
             }
         } // !useInkEdit
+
+        WindowHDC hdc(GetHWND());
+        m_richBaseDPI = wxSize(::GetDeviceCaps(hdc, LOGPIXELSX), ::GetDeviceCaps(hdc, LOGPIXELSY));
+        m_richUsedDPI = GetDPI();
```

Comment:
Sorry, could you please explain how could this return a different value? `GetDPI()` uses `WindowHDC` with the `HWND` of the top-level parent, surely it should use the same DPI as `m_richBaseDPI`? What am I missing?

Reply:
Imagine you have two displays. The main display with DPI 100% and a second display with DPI 150%. The system DPI will be 100% (96).
If you initialize the control on the second display, `m_richBaseDPI` will get value 96, and `m_richUsedDPI` will get the DPI of the second display (144).

Reply:
Wait, I see it now: you're using `WindowHDC(GetHWND())` here, but `m_hWnd` is still 0 because the window hadn't been created yet, so it's always the screen `HDC` for the primary screen in fact.

This is very, very confusing however (especially so when looking at the diff, which doesn't show that `MSWCreateControl()` is only called below) and I don't understand why would we do it like this instead of just using `ScreenHDC` directly.

Could you please change it like this or am I still missing something?

Reply:
I think you're still missing it. `GetDeviceCaps` will always return the system DPI, not matter on what display it is.` GetDPI` will use `GetDpiForWindow` and return the DPI of the display it (or its parent) is on.

Reply:
But using `GetHWND` here might not be the best thing to do. I see if I can move it to a better place.

Reply:
OK, thanks, I indeed forgot that `::GetDeviceCaps()` always used the main display values. But, still, if we need the main screen DPI, using `ScreenHDC` here would IMHO be more clear.